here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/connector/ConnectorInterfaceReadExecute.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017-2024 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.storage.http.connector;
+
+import static com.here.naksha.common.http.apis.ApiParamsConst.FEATURE_IDS;
+
+import com.here.naksha.lib.core.models.geojson.implementation.XyzFeatureCollection;
+import com.here.naksha.lib.core.models.naksha.Space;
+import com.here.naksha.lib.core.models.payload.Event;
+import com.here.naksha.lib.core.models.payload.events.feature.GetFeaturesByIdEvent;
+import com.here.naksha.lib.core.models.storage.ReadFeaturesProxyWrapper;
+import com.here.naksha.lib.core.models.storage.Result;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import com.here.naksha.storage.http.PrepareResult;
+import com.here.naksha.storage.http.RequestSender;
+import java.net.http.HttpResponse;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class ConnectorInterfaceReadExecute {
+
+  @NotNull
+  public static Result execute(ReadFeaturesProxyWrapper request, RequestSender sender) {
+    return switch (request.getReadRequestType()) {
+      case GET_BY_IDS -> executeFeaturesByIds(request, sender);
+      default -> throw new IllegalStateException("Unexpected value: " + request.getReadRequestType());
+    };
+  }
+
+  private static Result executeFeaturesByIds(ReadFeaturesProxyWrapper request, RequestSender sender) {
```

Comment:
There may be a need to extend Result object as well (for ex. nextPageToken) depending on what other fields that are to be returned as REST API response from Naksha Hub. However, this will depend on what is the difference in REST API response between Naksha Hub and Data Hub. For now, we can continue with current result object and can extend it when there is better clarity.